commec/setup.py

```diff
@@ -0,0 +1,529 @@
+\033[48;5;17m█ █████▀ █████▀ █ ▀█▄█▀                                 \033[38;5;202m      ████████▀         \033[0m
+                                                        \033[38;5;202m   ▄▄██████▀▀             \033[0m
+                                                        \033[38;5;202m ▀▀                       \033[0m""")
+        print("                 The Common Mechanism!",
+              "\n\nInternational Biosecurity and Biosafety Initiative for Science",
+              "\nCopyright © 2021-2024 ",
+              "\n\nThis script will help download the mandatory databases ",
+              "\nrequired for using Commec Screen, and requires a stable",
+              "\ninternet connection, wget, and update_blastdb.pl.",
+              "\n -> You can exit this setup at any time with \"exit\"",
+              "\n -> You can return to a previous step with \"back\"\n")
+
+        self.check_requirements()
+
+        self.setup_overall_directory()
+        self.do_setup()
+
+    def check_requirements(self):
+        """ Checks for wget, and update_blastdb.pl, which should both be present
+        given the conda environment to install and use Commec. """
+        print("Checking for wget, and update_blastdb")
+        dont_proceed = False
+        result = subprocess.run(["which", "wget"], check=False)
+        if result.returncode != 0:
+            print("Dependency wget could not be found. "
+                  "Check wget is installed in this environment.")
+            dont_proceed = True
+        result = subprocess.run(["which", "update_blastdb.pl"], check=False)
+        if result.returncode != 0:
+            print("Dependency update_blastdb.pl could not be found. "
+                  "Blast is likely not installed in this python environment.")
+            dont_proceed = True
+        if dont_proceed:
+            print("Error: Required dependency missing!")
+            self.stop()
+
+    def setup_overall_directory(self):
+        """
+        Get user inputs for global directory to store databases.
+        """
+        self.print_step(1)
+        user_input : str = ""
+        print("\nPlease provide the absolute or relative filepath",
+              "to where you would like the Commec databases to be located...",
+              "\nPress <Enter> to use default: ",
+              self.database_directory)
+        while True:
+            user_input : str = self.user_input()
+            if user_input == "back":
+                self.stop() # This is the first step, no point going back.
+            if len(user_input) > 0:
+                self.database_directory = user_input
+
+            try:
+                os.makedirs(self.database_directory, exist_ok=True)
+                try: # Removedirs can fail if it is a single leaf, containing other files.
+                    os.removedirs(self.database_directory)
+                except OSError:
+                    pass
```

Comment:
I don't quite follow this... are we checking that the path is writeable?

Could we factor out something like:
```
def can_write_to_directory(path: str) -> bool:
    """
    Validate if a directory can be created and is writeable.
    """
    os.makedirs(self.database_directory, exist_ok=True)
    return os.path.isdir(path) and os.access(path, os.W_OK)
  ```

Reply:
Great idea, makes it more readable for the purpose - I figured creating and destroying the folder / directory would suffice to check that it was a real directory.
Done.